src/cljs/athens/types/tasks/view.cljs

```diff
@@ -0,0 +1,336 @@
+                                                                                       :due-date         %}]))))
+
+            idle-fn            (gfns/debounce #(do
+                                                 (log/debug prop-name "idle-fn" (pr-str @local-value))
+                                                 (update-fn @local-value))
+                                              2000)
+            read-value         local-value
+            show-edit?         (r/atom true)
+            state-hooks        {:save-fn    save-fn
+                                :idle-fn    idle-fn
+                                :update-fn  update-fn
+                                :read-value read-value
+                                :show-edit? show-edit?}]
+        [:> FormControl {:is-required required?
+                         :is-invalid  invalid-prop-str?}
+         [:> FormLabel {:html-for prop-id}
+          prop-title]
+         [:> Box {:px           2
+                  :mt           2
+                  :minHeight    "2.125em"
+                  :borderRadius "sm"
+                  :bg           "background.attic"
+                  :cursor       "text"
+                  :_focusWithin {:shadow "focus"}}
+          ;; NOTE: we generate temporary uid for prop if it doesn't exist, so editor can work
+          [content-editor/block-content-el {:block/uid (or prop-block-uid
+                                                           ;; NOTE: temporary magic, stripping `:task/` 🤷‍♂️
+                                                           (str "tmp-" (subs prop-name
+                                                                             (inc (.indexOf prop-name "/")))
+                                                                "-uid-" (common.utils/gen-block-uid)))}
+           state-hooks]
+          [presence/inline-presence-el prop-block-uid]]
+
+         (if invalid-prop-str?
+           [:> FormErrorMessage
+            (str prop-title " is " (if required?
+                                     "required"
+                                     "empty"))]
+           [:> FormHelperText
+            (str "Please provide " prop-title)])]))))
+
+
+(defn- find-allowed-statuses
+  []
+  (let [task-status-page    (reactive/get-reactive-node-document [:node/title ":task/status"])
+        allowed-stat-blocks (-> task-status-page
+                                :block/properties
+                                (get ":property/enum")
+                                :block/children)
+        allowed-statuses    (map #(select-keys % [:block/uid :block/string]) allowed-stat-blocks)]
+    (if (seq allowed-statuses)
+      allowed-statuses
+      (let [[statuses ops] (create-default-allowed-statuses @db/dsdb)
+            event          (common-events/build-atomic-event ops)]
+        (rf/dispatch [:resolve-transact-forward event])
```

Comment:
This is where the prop enum is bootstraped. Later we should have a non-render level boostrapping, probably on boot.